### PR TITLE
fix: Allow use of hostname in meshNetworks gateways

### PR DESCRIPTION
**Pull request by mikebryant**
Part of #13581 - might well fix it?

### CODE REVIEW

File: pilot/pkg/bootstrap/server.go [hunk 1]
```diff
@@ -439,7 +439,8 @@ func (s *Server) initMeshNetworks(args *PilotArgs) error {
 		return nil
 	}
 	log.Infof("mesh networks configuration %s", spew.Sdump(meshNetworks))
-	util.ResolveHostsInNetworksConfig(s.meshNetworks)
+	util.ResolveHostsInNetworksConfig(meshNetworks)
```
Comment (howardjohn) [inline pilot/pkg/bootstrap/server.go]: So before we were resolving the hosts for the wrong meshNetworks config? Good catch

File: pilot/pkg/bootstrap/server.go [hunk 2]
```diff
@@ -439,7 +439,8 @@ func (s *Server) initMeshNetworks(args *PilotArgs) error {
...
+	log.Debugf("mesh networks configuration post-resolution %s", spew.Sdump(meshNetworks))
 	s.meshNetworks = meshNetworks
```
Comment (rshriram) [inline pilot/pkg/bootstrap/server.go]: Make this info please.

File: pilot/pkg/bootstrap/server.go [hunk 3]
```diff
@@ -452,7 +453,8 @@ func (s *Server) initMeshNetworks(args *PilotArgs) error {
 		}
 		if !reflect.DeepEqual(meshNetworks, s.meshNetworks) {
 			log.Infof("mesh networks configuration file updated to: %s", spew.Sdump(meshNetworks))
-			util.ResolveHostsInNetworksConfig(s.meshNetworks)
+			util.ResolveHostsInNetworksConfig(meshNetworks)
+			log.Debugf("mesh networks configuration post-resolution %s", spew.Sdump(meshNetworks))
 			s.meshNetworks = meshNetworks
```
Comment (rshriram) [inline pilot/pkg/bootstrap/server.go]: Same as above. Infof

File: pilot/pkg/networking/util/util.go [hunk 4]
```diff
@@ -264,6 +264,9 @@ func ResolveHostsInNetworksConfig(config *meshconfig.MeshNetworks) {
 			gwIP := net.ParseIP(gw.GetAddress())
 			if gwIP == nil {
 				addrs, err := net.LookupHost(gw.GetAddress())
+				if err != nil {
+					log.Warnf("error resolving host %#v: %v", gw.GetAddress(), err)
+				}
```
Comment (rshriram) [inline pilot/pkg/networking/util/util.go]: Is this called only from bootstrap? Else it will flood the logs.

Comment (hzxuzhonghu) [reply pilot/pkg/networking/util/util.go]: Also called when filewatcher detects file change

Comment (mikebryant) [reply pilot/pkg/networking/util/util.go]: I think those two cases are fine - I wouldn't expect the meshNetworks config file to be changing often. And when it does - I'd like to see if host resolution fails then too

File: pilot/pkg/bootstrap/server.go (hunk 2 above)
Comment (mikebryant) [reply pilot/pkg/bootstrap/server.go]: Done :)

File: pilot/pkg/bootstrap/server.go (hunk 3 above)
Comment (mikebryant) [reply pilot/pkg/bootstrap/server.go]: Done :)

**Comment by mikebryant** [comment]
I'm not sure why the tests are failing - I'd appreciate some help with that

**Comment by mikebryant** [comment]
Cool, tests passed :).

Is there anything else I need to do for this PR? :)

**Pull request merged**